node/libs/storage/src/traits.rs

```diff
@@ -45,53 +45,31 @@ pub trait BlockStore: fmt::Debug + Send + Sync {
     fn subscribe_to_block_writes(&self) -> watch::Receiver<BlockNumber>;
 }
 
-#[async_trait]
```

Comment:
I don't see why we should treat Arc specially here (i.e. why not Box, Rc, &, &mut, etc.). Also I lost a significant amount of time debugging why my implementation of verify_payload on X is overridden by the default implementation on Arc<X> (kind of my fault, but I'm not taking the full blame here ;P). As you say, it can always be wrapped in a newtype anyway.